GsaGHTests/Helpers/Assemble/AssembleModelLists.cs

```diff
@@ -0,0 +1,64 @@
+﻿using GsaGH.Components;
+using GsaGH.Parameters;
+
+using Xunit;
+
+namespace GsaGHTests.Helpers.Assemble {
+  [Collection("GrasshopperFixture collection")]
+  public class AssembleModelLists {
+    private const int DefaultId = 1;
+    private const string ExpectedName = "create list test";
+    private const string ExpectedDefinition = "1 2 3";
+
+    [Fact]
+    public void ListShouldReturnDefinitionWhenIdIsSet() {
+      const int ExpectedId = 2;
+      GsaList gsaList = GsaList(ExpectedId);
+
+      Assert.Equal(ExpectedId, gsaList.Id);
+      Assert.Equal(ExpectedName, gsaList.Name);
+      Assert.Equal(ExpectedDefinition, gsaList.Definition);
+    }
+
+    [Fact]
+    public void ListShouldReturnDefinitionWhenIdIsNotSet() {
+      GsaList gsaList = GsaList();
+
+      Assert.Equal(DefaultId, gsaList.Id);
+      Assert.Equal(ExpectedName, gsaList.Name);
+      Assert.Equal(ExpectedDefinition, gsaList.Definition);
+    }
+
+    private static GsaList GsaList(int id = 1) {
+      GsaListGoo listComponent = CreateListComponent(id);
+      GsaModelGoo gsaModelGoo = CreateModelComponent(listComponent);
+      return gsaModelGoo.Value.GetLists()[0];
+    }
+
+    private static GsaListGoo CreateListComponent(int id) {
+      var createListComponent = new CreateList();
+      createListComponent.CreateAttributes();
+
+      if (id != DefaultId) {
+        ComponentTestHelper.SetInput(createListComponent, id, 0);
+      }
```

Comment:
if you set id to `null` to one of your tests that check if the default will be set. will it achieve the same effect, without the need for the conditional? line 25 `ListShouldReturnDefinitionWhenIdIsNotSet` I think.